src/Compilers/CSharp/Portable/Syntax/SyntaxEquivalence.cs

```diff
@@ -100,130 +101,147 @@ private static bool AreTokensEquivalent(GreenNode? before, GreenNode? after, Fun
 
         private static bool AreEquivalentRecursive(GreenNode? before, GreenNode? after, Func<SyntaxKind, bool>? ignoreChildNode, bool topLevel)
         {
-            if (before == after)
-            {
-                return true;
-            }
+            // Use an explicit stack so we can walk down deep trees without blowing the real stack.
```

Comment:
>Use an explicit stack [](http://example.com/codeflow?start=15&length=21)

Could we add a test that would previously fail? (Say, two trees with deeply nested binary operators, either equivalent or distinct at the deepest token.)

Reply:
done.